node/light.go

```diff
@@ -38,10 +39,15 @@ type LightNode struct {
 func newLightNode(
 	conf config.Config,
 	genesis genesis.Genesis,
-	p2pClient *p2p.Client,
+	nodeKey *key.NodeKey,
 	database ds.Batching,
 	logger zerolog.Logger,
 ) (ln *LightNode, err error) {
+	p2pClient, err := p2p.NewClient(conf.P2P, nodeKey.PrivKey, database, genesis.ChainID, logger, nil)
```

Comment:
what is reasoning behind moving this from the composing part to the constructor?